projects/ngrx.io/content/guide/entity/adapter.md

```diff
@@ -137,41 +137,42 @@ export const initialState: State = adapter.getInitialState({
   selectedUserId: null,
 });
 
-export const reducer = createReducer(initialState,
-  on(UserActions.addUser, state => {
-   return adapter.addOne(action.user, state)
+export const reducer = createReducer(
+  initialState,
+  on(UserActions.addUser, (state, { user }) => {
+    return adapter.addOne(user, state)
   }),
-  on(UserActions.upsertUser, state => {
-    return adapter.upsertOne(action.user, state);
+  on(UserActions.upsertUser, (state, { user }) => {
```

Comment:
So it will be something line UserActions.payload right, and then payload.user.

Reply:
> I think for the docs it would be clearer if we don't deconstruct and use action.user?

I'd rather deconstruct (because we can do that now 😃) and add the comment that it's action that is deconstructed.
Also, we frequently deconstruct in Effect, I don't think this is any different 🙂

Reply:
Ok let me know what is final, so I can make the changes. :)

Reply:
Let's do deconstruction with the comments above it (in a few places). I feel like the code in the examples should be as real-life as possible. 😃
@brandonroberts agrees as well.

Reply:
Done.